### PR TITLE
Switched from RLInterface to CommonRLInterface

### PR DESCRIPTION
**Pull request by zsunberg**
I switched from RLInterface to CommonRLInterface, so we should be able to register this. It is also almost possible to use the package with a CommonRLInterface.AbstractEnv, but not quite (see #51 and #52). I am not sure if there are any performance regressions.

### CODE REVIEW

File: src/evaluation_policy.jl [hunk 1]
```diff
@@ -7,25 +7,28 @@ Interface for defining an evaluation policy
     returns the average reward of the current policy, the user can specify its own function 
     f to carry the evaluation, we provide a default basic_evaluation that is just a rollout. 
 """
-function evaluation(f::Function, policy::AbstractNNPolicy, env::AbstractEnvironment, n_eval::Int64, max_episode_length::Int64, verbose::Bool = false)
+function evaluation(f::Function, policy::AbstractNNPolicy, env::AbstractEnv, n_eval::Int64, max_episode_length::Int64, verbose::Bool = false)
```
Comment (rejuvyesh) [inline src/evaluation_policy.jl:10]: How did `^M` come in? Likely a `vim` issue?

File: src/solver.jl [hunk 2]
```diff
@@ -28,16 +28,16 @@
...
 
 function POMDPs.solve(solver::DeepQLearningSolver, problem::MDP)
-    env = MDPEnvironment(problem, rng=solver.rng)
+    env = MDPCommonRLEnv{AbstractArray{Float32}}(problem) # ignores solver.rng because CommonRLEnv doesn't have rng support yet
     return solve(solver, env)
 end
 
 function POMDPs.solve(solver::DeepQLearningSolver, problem::POMDP)
-    env = POMDPEnvironment(problem, rng=solver.rng)
+    env = POMDPCommonRLEnv{AbstractArray{Float32}}(problem) # ignores solver.rng because CommonRLEnv doesn't have rng support yet
```
Comment (rejuvyesh) [inline src/solver.jl:36]: We can't get any more concrete information except `AbstractArray{Float32}`?

**Review by rejuvyesh — CHANGES_REQUESTED** [review]
Looks good to me except for some small issues I have pointed out.

File: src/evaluation_policy.jl (hunk 1 above)
Comment (zsunberg) [reply src/evaluation_policy.jl:10]: Yeah, I am not sure how the ^M is happening. I will correct it. Thanks

File: src/solver.jl (hunk 2 above)
Comment (zsunberg) [reply src/solver.jl:36]: I think `AbstractArray{Float32}` is what we want here. It just means that `convert_o(AbstractArray{Float32}, o, pomdp)` will be called on every observation. That way problem-writers can use static arrays or built-in arrays. If the problem implementation is type-stable, the compiler should still be able to infer a nice concrete return type.

**Comment by zsunberg** [comment]
@MaximeBouton do you have any comments on or concerns about this?

**Review by rejuvyesh — APPROVED** [review]
LGTM! :shipit:

**Comment by MaximeBouton** [comment]
Sorry I took a lot of time off :) 
Reviewing this right now

**Comment by MaximeBouton** [comment]
That looks good! We can merge it, should we tag a version 0.6 and register?

**Comment by zsunberg** [comment]
Yes, I think we can register! Go ahead or I can do it in the next few days

**Pull request merged**